tokio-util/src/sync/cancellation_token/tree_node.rs

```diff
@@ -178,7 +178,7 @@ where
                 locked_node = node.inner.lock().unwrap();
                 locked_parent
             }
-            Err(TryLockError::Poisoned(err)) => Err(err).unwrap(),
+            Err(TryLockError::Poisoned(err)) => panic!("{:?}", err),
```

Comment:
Hmm. I remember writing this line of code. I actually used `unwrap` here because I thought it would lead to less code generation, as it could just call into the same code as all of the other places where `PoisonError` is unwrapped.

Thoughts?

Reply:
Hmm, indeed it may be preferable from codegen's point of view. I have reverted this change with a link to your comment.